courses/free-courses-en.md

```diff
@@ -223,6 +223,8 @@
 
 ### C
 
+* [C & Data Structure \| Depth understanding](https://www.youtube.com/watch?v=4OGMB4Fhh50&list=PLBlnK6fEyqRhX6r2uhhlubuF5QextdCSM) - Neso Academy
```

Comment:
For youtube, we prefer "playlist" links and the name of the producer/creator. https://github.com/EbookFoundation/free-programming-books/blob/master/CONTRIBUTING.md#platforms-and-access-notes

Reply:
I made the change